src/machine/image-varlink.c

```diff
@@ -17,24 +17,17 @@ typedef struct ImageUpdateParameters {
 } ImageUpdateParameters;
 
 int vl_method_update_image(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
-        struct params {
-                const char *image_name;
-                const char *new_name;
-                int read_only;
-                uint64_t limit;
-        };
-
         static const sd_json_dispatch_field dispatch_table[] = {
-                { "name",     SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(struct params, image_name), SD_JSON_MANDATORY },
-                { "newName",  SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(struct params, new_name),   0 },
-                { "readOnly", SD_JSON_VARIANT_BOOLEAN,       sd_json_dispatch_tristate,     offsetof(struct params, read_only),  0 },
-                { "limit",    _SD_JSON_VARIANT_TYPE_INVALID, sd_json_dispatch_uint64,       offsetof(struct params, limit),      0 },
+                { "name",     SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(ImageUpdateParameters, name),      SD_JSON_MANDATORY },
+                { "newName",  SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(ImageUpdateParameters, new_name),  0                 },
+                { "readOnly", SD_JSON_VARIANT_BOOLEAN,       sd_json_dispatch_tristate,     offsetof(ImageUpdateParameters, read_only), 0                 },
+                { "limit",    _SD_JSON_VARIANT_TYPE_INVALID, sd_json_dispatch_uint64,       offsetof(ImageUpdateParameters, limit),     0                 },
                 VARLINK_DISPATCH_POLKIT_FIELD,
                 {}
         };
 
         Manager *manager = ASSERT_PTR(userdata);
-        struct params p = {
+        ImageUpdateParameters p ={
                 .read_only = -1,
                 .limit = UINT64_MAX,
         };
```

Comment:
i think it would make sense to introduce a macro:

```c
#define IMAGE_UPDATE_PARAMETERS_NULL \
        (ImageUpdateParamaters) { \
                .read_only = -1, \
                .limit = UINT64_MAX, \
         }
```

and then use that wherever we initialize the struct. nicely generically. (we have a similar pattern elsewhere)